### PR TITLE
[MS] Add C compilers and disable python

### PR DESCRIPTION
**Pull request by partouf**
Would fix https://github.com/compiler-explorer/compiler-explorer/issues/1371

Based on mstodo branch.

Notes on this change:
* Basically just adds the same msvc compilers to c.local.properties with a prefix added to compiler id names
* Libraries are left out, as they are mostly/all c++ - when we add C libraries like for example OpenSSL we can worry about that then.

### CODE REVIEW

File: windows/Msvce.psm1 [hunk 1]
```diff
@@ -1464,6 +1479,15 @@ function Install-MsvceConfigurationFile {
     [string]$SdkVersion
   )
 
+  $compilerIdPrefix = ''
+  $propertiesFilename = 'c++.local.properties'
+
+  if ($GenerateClocal -eq $true) {
```
Comment (strega-nil) [inline windows/Msvce.psm1:1485]:
```suggestion
  if ($CProperties) {
```

File: windows/Msvce.psm1 [hunk 2]
```diff
@@ -1454,6 +1466,9 @@ function Install-MsvceConfigurationFile {
     [Parameter(Mandatory=$true, ParameterSetName='FindSdk')]
     [string]$DockerTag,
 
+    [Parameter(Mandatory=$true)]
+    [bool]$GenerateClocal,
```
Comment (strega-nil) [inline windows/Msvce.psm1:1470]:
```suggestion
    [switch]$CProperties,
```

File: windows/Msvce.psm1 [hunk 3]
```diff
@@ -1464,6 +1479,15 @@ function Install-MsvceConfigurationFile {
     [string]$SdkVersion
   )
 
+  $compilerIdPrefix = ''
+  $propertiesFilename = 'c++.local.properties'
+
+  if ($GenerateClocal -eq $true) {
+    $compilerIdPrefix = 'c';
+    $propertiesFilename = 'c.local.properties'
+  }
```
Comment (strega-nil) [inline windows/Msvce.psm1:1488]:
```suggestion
  } else {
      $compilerIdPrefix = ''
      $propertiesFilename = 'c++.local.properties'
  }
```

File: windows/Msvce.psm1 [hunk 4]
```diff
@@ -1464,6 +1479,15 @@ function Install-MsvceConfigurationFile {
     [string]$SdkVersion
   )
 
+  $compilerIdPrefix = ''
+  $propertiesFilename = 'c++.local.properties'
```
Comment (strega-nil) [inline windows/Msvce.psm1:1483]:
```suggestion
```

File: windows/Msvce.psm1 [hunk 5]
```diff
@@ -1617,28 +1641,30 @@ function Install-MsvceConfigurationFile {
     $_.Name -replace '-','_'
   }
 
-  $file += ('libs=' + ($libraryNames -join ':'))
+  if ($GenerateClocal -eq $false) {
```
Comment (strega-nil) [inline windows/Msvce.psm1:1644]:
```suggestion
  if (-not $CProperties) {
```

File: windows/Msvce.psm1 [hunk 6]
```diff
@@ -828,11 +828,19 @@ function Build-MsvceDataDirectory {
   Write-Verbose 'Installing MSVCE base CE files'
   Install-MsvceBaseCEFiles -DataDirectory $DataDirectory
 
-  Write-Verbose 'Installing MSVCE configuration file'
+  Write-Verbose 'Installing MSVCE c++ configuration file'
   Install-MsvceConfigurationFile `
     -DataDirectory $DataDirectory `
     -DockerTag $DockerTag `
+    -GenerateClocal $false `
```
Comment (strega-nil) [inline windows/Msvce.psm1:835]:
```suggestion
```

File: windows/Msvce.psm1 [hunk 7]
```diff
@@ -828,11 +828,19 @@ function Build-MsvceDataDirectory {
...
     | Out-Null
+
+  Write-Verbose 'Installing MSVCE c configuration file'
+  Install-MsvceConfigurationFile `
+    -DataDirectory $DataDirectory `
+    -DockerTag $DockerTag `
+    -GenerateClocal $true `
```
Comment (strega-nil) [inline windows/Msvce.psm1:842]:
```suggestion
    -CProperties `
```

File: windows/Msvce.psm1 [hunk 8]
```diff
@@ -828,11 +828,19 @@ function Build-MsvceDataDirectory {
   Write-Verbose 'Installing MSVCE base CE files'
   Install-MsvceBaseCEFiles -DataDirectory $DataDirectory
 
-  Write-Verbose 'Installing MSVCE configuration file'
+  Write-Verbose 'Installing MSVCE c++ configuration file'
```
Comment (strega-nil) [inline windows/Msvce.psm1:831]:
```suggestion
  Write-Verbose 'Installing MSVCE C++ configuration file'
```

File: windows/Msvce.psm1 [hunk 9]
```diff
@@ -828,11 +828,19 @@ function Build-MsvceDataDirectory {
...
   Install-MsvceConfigurationFile `
     -DataDirectory $DataDirectory `
     -DockerTag $DockerTag `
+    -GenerateClocal $false `
     | Out-Null
+
+  Write-Verbose 'Installing MSVCE c configuration file'
```
Comment (strega-nil) [inline windows/Msvce.psm1:838]:
```suggestion
  Write-Verbose 'Installing MSVCE C configuration file'
```

File: windows/Msvce.psm1 [hunk 10]
```diff
@@ -1418,6 +1426,10 @@ The directory which MSVCE can use as its C:\Data.
 
 The tag of the msvce image to look in, to see where the SDK is installed.
 
+.PARAMETER GenerateCLocal
+
+If true, generate a c.local.properties file instead of for c++.
```
Comment (strega-nil) [inline windows/Msvce.psm1:1431]:
```suggestion
.PARAMETER CProperties

Generate c.local.properties as opposed to c++.local.properties.
```

File: windows/Msvce.psm1 [hunk 11]
```diff
@@ -1454,6 +1466,9 @@ function Install-MsvceConfigurationFile {
     [Parameter(Mandatory=$true, ParameterSetName='FindSdk')]
     [string]$DockerTag,
 
+    [Parameter(Mandatory=$true)]
```
Comment (strega-nil) [inline windows/Msvce.psm1:1469]:
```suggestion
    [Parameter()]
```

**Review by strega-nil — CHANGES_REQUESTED** [review]

**Comment by partouf** [comment]
Thanks for the review @strega-nil !
Suggestions make sense, will apply and test tomorrow.

**Comment by partouf** [comment]
merged everything, worked as advertised

**Pull request merged**